lib/event_store_client/store_adapter/in_memory.rb

```diff
@@ -13,16 +19,21 @@ def append_to_stream(stream_name, events, expected_version: nil) # rubocop:disab
             'eventId' => event.id,
             'data' => event.data,
             'eventType' => event.type,
-            'metadata' => event.metadata,
+            'metaData' => event.metadata,
             'positionEventNumber' => event_store[stream_name].length
           )
         end
       end
 
-      def read(stream_name, direction: 'forward', start: 0, count: per_page, resolve_links: nil)
-        read_stream_forward(stream_name, start: start, count: count) if direction == 'forward'
+      def read(stream_name, direction: 'forward', start: 0, resolve_links: nil)
+        response =
```

Comment:
what about use here `send("read_stream_#{direction}", stream_name, start: start)`

Reply:
In my opinion, your suggestion is ok for a case when there are more than 2 methods. You have to check that an object responds to a method.